### PR TITLE
Login and signup layouts complete with progress on login logic

### PR DESCRIPTION
**Pull request by s-kettle**
Lots of work done here. Draft layouts complete for logging in and signing up. Made progress with using Retrofit for a login request. Backend team may need to look at the data objects to see what's sent/expected. Whoever is working on the login endpoint, perhaps we can catch up on this on Mon.

### CODE REVIEW

File: app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt [hunk 1]
```diff
@@ -4,20 +4,15 @@ import retrofit2.Retrofit
...
-        return retrofit!!.create(ListingAPIService::class.java)
+    val instance: APIService by lazy {
+        val retrofit = Retrofit.Builder()
+            .baseUrl(baseUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        retrofit.create(APIService::class.java)
     }
-
 }
```
Comment (Lonrot) [inline app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt:18]: Hi, are we not adhering to the singleton pattern? I guess, if we make call during request we will make more instances of the class

File: app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt [hunk 2]
```diff
@@ -4,20 +4,15 @@ import retrofit2.Retrofit
...
-                .addConverterFactory(GsonConverterFactory.create())
-                .build()
-        }
-        return retrofit!!.create(ListingAPIService::class.java)
+    val instance: APIService by lazy {
+        val retrofit = Retrofit.Builder()
+            .baseUrl(baseUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        retrofit.create(APIService::class.java)
```
Comment (Lonrot) [inline app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt:16]: Hi, are we not adhering to the singleton pattern? I guess, if we make call during request we will make more instances of the class

**Review by Lonrot — CHANGES_REQUESTED** [review]
It all fantastic, The only thing to note is about the retrofit instance.

File: app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt (hunk 2 above)
Comment (s-kettle) [reply app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt:16]: Hi Angel. In Kotlin, an "object" is a singleton. Note that the file is defined as object and not class. Hope that makes sense.

Comment (Lonrot) [reply app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt:16]: Hi Sam , all clear and well explained. Seems like Kotlin is a few steps ahead of Java

**Review by Lonrot — APPROVED** [review]

**Pull request merged**